molgenis-data-annotators/src/main/java/org/molgenis/data/annotation/resources/impl/MultiFileResource.java

```diff
@@ -0,0 +1,83 @@
+package org.molgenis.data.annotation.resources.impl;
+
+import org.molgenis.data.Entity;
+import org.molgenis.data.EntityMetaData;
+import org.molgenis.data.Query;
+import org.molgenis.data.QueryRule;
+import org.molgenis.data.annotation.resources.MultiResourceConfig;
+import org.molgenis.data.annotation.resources.Resource;
+import org.molgenis.data.annotation.resources.ResourceConfig;
+import org.molgenis.data.vcf.VcfRepository;
+
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * Created by charbonb on 15/06/15.
+ */
+public class MultiFileResource implements Resource
+{
+	private String name;
+	private Map<String, Resource> resources;
+
+	public MultiFileResource(String name, MultiResourceConfig config, EntityMetaData emd)
+	{
+		this.name = name;
+		config.getConfigs();
+		Map<String, Resource> resources = new HashMap<>();
+		Map<String, ResourceConfig> configs = config.getConfigs();
+		for (String chrom : configs.keySet())
+		{
+			resources.put(chrom, new ResourceImpl(name + chrom, configs.get(chrom), new TabixRepositoryFactory(emd)));
```

Comment:
I'd add the RepositoryFactory instead in the constructor, then it works for both Tabix and TabixVcf